packages/block-library/src/image/view.js

```diff
@@ -325,7 +326,11 @@ const { state, actions, callbacks } = store(
 			setButtonStyles() {
 				const ctx = getContext();
 				const { ref } = getElement();
-				ctx.imageRef = ref;
+				state.metadata[ ctx.imageId ] = {
+					...state.metadata[ ctx.imageId ],
+					imageRef: ref,
+					currentSrc: ref.currentSrc,
+				};
```

Comment:
In the Interactivity API, immutability isn't necessary. In fact, since we are using signals, it's more performant to use mutability.

Reply:
Done in https://github.com/WordPress/gutenberg/pull/63348/commits/3c3c3f27ea50e8c8749b8bd2e41a47afdc598142.